commons/ava_networks/test_gecko_network.go

```diff
@@ -64,26 +64,37 @@ type TestGeckoNetworkServiceConfig struct {
 	// Whether the certs used by services with this configuration will be different or not
 	varyCerts bool
 	serviceLogLevel ava_services.GeckoLogLevel
+	// Alternate image name, if not the default Gecko image.
+	// Used primarily for Byzantine tests but can also test heterogenous Gecko versions, for example.
+	alternateImageName *string
+	snowQuorumSize     int
+	snowSampleSize     int
 }
 
 func NewTestGeckoNetworkServiceConfig(
 			varyCerts bool,
-			serviceLogLevel ava_services.GeckoLogLevel) *TestGeckoNetworkServiceConfig {
+			serviceLogLevel ava_services.GeckoLogLevel,
+			alternateImageName *string,
```

Comment:
Rather than calling this `alternate...` why not just call it `imageName`? That way, the user is always responsible for declaring which image they want (be it the regular image, or the Byzantine image)

Reply:
Also, separate comment: is there any reason this is `*string` rather than just `string`? IMO we should pass everything by-val unless we explicitly need by-ref, which protects us from weirdness and makes us think consciously about using pointers.